management_api_app/db/repositories/workspace_templates.py

```diff
@@ -0,0 +1,43 @@
+from typing import List
+
+from azure.cosmos import CosmosClient
+
+from core import config
+from db.errors import EntityDoesNotExist
+from db.repositories.base import BaseRepository
+from models.domain.resource_template import ResourceTemplate
+
+
+class WorkspaceTemplateRepository(BaseRepository):
+    def __init__(self, client: CosmosClient):
+        super().__init__(client, config.STATE_STORE_RESOURCE_TEMPLATES_CONTAINER)
+
+    @staticmethod
+    def _workspace_template_by_name_query(name: str) -> str:
+        return f'SELECT * FROM c WHERE c.resourceType = "workspace" AND c.name = "{name}"'
+
+    def get_workspace_templates_by_name(self, name: str) -> List[ResourceTemplate]:
+        query = self._workspace_template_by_name_query(name)
+        return self._query(query=query)
+
+    def get_current_workspace_template_by_name(self, name: str) -> dict:
+        query = self._workspace_template_by_name_query(name) + ' AND c.isCurrent = true'
+        workspace_templates = self._query(query=query)
+        if len(workspace_templates) != 1:
```

Comment:
if len == 0 then the exception holds true. But if len > 1 then there should be a different exception. That should never happen in correctly functioning system but it might. When we receive a new bundle registration we should supercede the last entry and mark the new one as current. That could fail and we might get two current. That is an error and we should report an assert on it.